controllers/resources.go

```diff
@@ -1144,7 +1144,7 @@ func (r *DurosReconciler) deleteResourceWithWait(ctx context.Context, log logr.L
 	}
 
 	log.Info("cleaning up resource", "name", resource.Key.Name, "namespace", resource.Key.Namespace)
-	return wait.PollImmediateInfiniteWithContext(ctx, 100*time.Millisecond, func(context.Context) (done bool, err error) {
+	return wait.PollUntilContextCancel(ctx, 100*time.Millisecond, true, func(context.Context) (done bool, err error) {
```

Comment:
@Gerrit91 PollImmediateInfiniteWithContext is deprecated now, but the deprecation warning suggests a non-existing replacement, tried my best to understand what the original intention was and took the best matching non-deprecated poll implementation

Reply:
Should be good.